js/packages/teams-ai/src/internals/OpenAIClient.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @module teams-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// eslint-disable-next-line import/named
+import axios, { AxiosInstance } from 'axios';
+import {
+    CreateChatCompletionRequest,
+    CreateChatCompletionResponse,
+    CreateCompletionRequest,
+    CreateCompletionResponse,
+    CreateEmbeddingRequest,
+    CreateEmbeddingResponse,
+    CreateModerationRequest,
+    ModerationResponse
+} from './types';
+
+/**
+ * @private
+ */
+export interface OpenAIClientResponse<TData> {
+    status: number;
+    statusText: string;
+    headers: Record<string, string>;
+    data?: TData;
+}
+
+/**
+ * @private
+ */
+export interface OpenAIClientOptions {
+    apiKey: string;
+    organization?: string;
+    endpoint?: string;
+    headerKey?: string;
+}
+
+/**
+ * @private
+ */
+export class OpenAIClient {
+    private _httpClient: AxiosInstance;
+
+    private readonly DefaultEndpoint = 'https://api.openai.com';
+    private readonly UserAgent = 'Microsoft Teams Conversational AI SDK';
```

Comment:
We should have a group discussion to align on 'official user agent strings' for consistency :) non-blocking.

Reply:
good catch